db/dataset.py

```diff
@@ -315,3 +355,147 @@ def _delete_snapshots_for_dataset(connection, dataset_id):
+    """Adds new recordings to a class in a dataset.
+
+    If any given recording already exist in this class it is not added again.
+
+    Args:
+        dataset_id: the uuid of the dataset
+        class_name: The class to add the recordings to
+        recordings: the recordings to add
+
+    Raises:
+        NoDataFoundException if the dataset doesn't exist or if the class doesn't exist in the dataset
+    """
+
+    with db.engine.begin() as connection:
+        clsid = _get_classid_for_dataset(connection, dataset_id, class_name)
+        for mbid in recordings:
+            connection.execute(sqlalchemy.text("""
+              INSERT INTO dataset_class_member (class, mbid)
+                   SELECT :clsid, :mbid
+         WHERE NOT EXISTS (SELECT *
+                             FROM dataset_class_member d
+                            WHERE d.class = :clsid
+                              AND d.mbid = :mbid)
+                      """),
+                               {"clsid": clsid, "mbid": mbid})
+
+
+def delete_recordings(dataset_id, class_name, recordings):
+    """Delete recordings from a dataset class"""
+
+    with db.engine.begin() as connection:
+        clsid = _get_classid_for_dataset(connection, dataset_id, class_name)
+        for mbid in recordings:
+            connection.execute(sqlalchemy.text("""
+                DELETE FROM dataset_class_member
+                      WHERE class = :class_name
+                        AND mbid = :mbid_num
+                            """),
+                               {"class_name": clsid, "mbid_num": mbid})
+
+
+def add_class(dataset_id, dict):
+    """Add a class to a dataset
+
+    If the dict argument contains a key "recordings", add these recordings
+    to the class as well. See :func:`add_recordings`.
+    If the class already exists, do not add it again, but add recordings
+    to it if they are given.
+
+    Args:
+        dataset_id: the UUID of the dataset to add this class to
+        dict: A dictionary representing the class to add:
+              {"name": "Classname", "description": "Class desc",
+               "recordings": [list of recording ids (optional)}
+    """
+
+    with db.engine.begin() as connection:
+        if "description" not in dict:
+            dict["description"] = None
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO dataset_class (name, description, dataset)
+                 SELECT :name, :description, :datasetid
+       WHERE NOT EXISTS (SELECT *
+                           FROM dataset_class d
+                          WHERE d.name = :name
+                            AND d.dataset = :datasetid)
+                    """),
+                           {"name": dict["name"], "description": dict["description"], "datasetid": dataset_id})
+    if "recordings" in dict:
+        add_recordings(dataset_id, dict["name"], dict["recordings"])
+
+
+def delete_class(dataset_id, dict):
```

Comment:
Again, better not use `dict`. In this case it can just be replaced with a `class_name`.